worker/worker.py

```diff
@@ -522,8 +539,10 @@ def main():
 
     print("Waiting for the heartbeat thread to finish...")
     heartbeat_thread.join()
-    print("Finishing the worker normally")
-
+    
+    return 0 if fish_exit else 1
 
 if __name__ == "__main__":
-    main()
+    ret=worker()
```

Comment:
If I'm not wrong `sys.exit(worker())` should made the work, see
https://softwareengineering.stackexchange.com/questions/418600/best-practice-for-python-main-function-definition-and-program-start-exit